tcl/filter.tcl

```diff
@@ -50,10 +50,13 @@ proc qc::filter_authenticate {event {error_handler qc::error_handler}} {
         # Check if this request is registered.
         if { [qc::registered $method $url_path] } {
             if {[qc::cookie_exists session_id] && [qc::session_valid [qc::session_id]]} {
-                qc::session_update [qc::session_id]
+                global session_id current_user_id
+                qc::session_update $session_id
+                set current_user_id [qc::session_user_id $session_id]
             } elseif {$method ni [list "GET" "HEAD"] && [qc::cookie_exists session_id] && ! [qc::session_valid [qc::session_id]]} {
```

Comment:
Is setting the current_user_id necessary here, would have thought the first time qc::auth is called this would be handled for us. Or am I missing something ?